### PR TITLE
[Squid][SMB][Automation]Deploy samba with 500 shares using declarative

### PR DESCRIPTION
**Pull request by MohitBis**
[Squid][SMB][Automation]Deploy samba with 500 shares using declarative style

### CODE REVIEW

**Comment by MohitBis** [comment]
Log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-3EVFUO

**Review by pranavprakash20 — DISMISSED** [review]
LGTM

File: suites/squid/smb/tier-1-smb-500-shares-declarative.yaml [hunk 1]
```diff
@@ -0,0 +1,4100 @@
...
+                - {name: user1, password: passwd}
+              groups: []
+          - resource_type: ceph.smb.share
+            cluster_id: smb1
+            share_id: share1
+            cephfs:
+              volume: cephfs
+              subvolumegroup: smb
+              subvolume: sv1
+              path: /
```
Comment (psathyan) [inline suites/squid/smb/tier-1-smb-500-shares-declarative.yaml:108]: IMHO, we should use a single block that accepts the number of shares to be created as input / config. Internally, we  should create the necessary test data specs required.

This is to large and will be difficult to maintain and scale.

Comment (MohitBis) [reply suites/squid/smb/tier-1-smb-500-shares-declarative.yaml:108]: HI @psathyan 

We are already doing it in other suite (Already merged)
https://github.com/red-hat-storage/cephci/blob/master/suites/squid/smb/tier-1-smb-500-shares.yaml


This is the ask from smb dev team

**Review by tintumathew10 — COMMENTED** [review]
LGTM

**Review by adityaramteke — APPROVED** [review]
LGTM

**Comment by MohitBis** [comment]
Latest Log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-36ZRMZ

**Review by pranavprakash20 — APPROVED** [review]
Lgtm

**Review by adityaramteke — APPROVED** [review]
LGTM

**Review by psathyan — COMMENTED** [review]
/lgtm

File: suites/squid/smb/tier-1-smb-500-shares-declarative.yaml (hunk 1 above)
Comment (psathyan) [reply suites/squid/smb/tier-1-smb-500-shares-declarative.yaml]: Thanks @MohitBis for addressing the comment. Nice to see you are reusing the modules.

**Review by tintumathew10 — APPROVED** [review]

**Pull request merged**